### PR TITLE
Updated display according to user default theme

### LINKED ISSUE #47: Feature: implement display according to user default system theme
### Description:
Currently, the app lacks support for adapting its display to the user's system theme preference. To enhance user experience and align with user expectations, we should implement the ability to automatically switch between light and dark themes based on the user's system-wide preference and also add it to settings to change manually

### Tips for the issue:
- You can edit the icon on top to write and implement it.
- Feel free to change the UI
- Making a settings page screen and implementing it there is my recommendation

### To do:
- [x] Ask us to assign the issue
- [x] Once the issue is assigned, you can start working on it
- [x] Create a PR

### Resource:
- [Flutter docs](https://docs.flutter.dev) 

### Notes:
The task is **assigned to contributors on a first-come, first-serve basis**, and the **contributor has to report progress every 3 days** to ensure active development on the tasks.

### PR DESCRIPTION
**Pull request by 0kt1**
### Issue:
[Implement display according to user default system theme ](https://github.com/IEEE-VIT/hacktoberfest-flutter/issues/47)

### Changes made
-   Added a settings page screen to change the theme.
-   Updated the themes for the app.
-   Made some UI changes.

### How does the solution address the problem
This PR updates the app to follow the system's default theme and offers users the choice of light, dark, or system default themes.

### Linked issues
Resolves #47

### CODE REVIEW

**Comment by dk-a-dev** [comment]
@0kt1 please try to pull from master and resolve merge conflicts ASAP
Your PR is under review

**Comment by dk-a-dev** [comment]
We have added new lint rules which will produce a lot of warnings try to first resolve conflicts and then refactor a/c to warnings, if u need my help message me, then I will continue ur PR review
@0kt1

File: lib/screens/settings.dart [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                                  device.theme = e;
+                                });
+                              },
+                            )).toList(),
+                          ),
+                        ),
+                        actions: [
+                          TextButton(onPressed: () {
+                               Navigator.of(context).pop();
+                               Provider.of<ThemeProvider>(context, listen: false).changeTheme(device.theme);
```
Comment (dk-a-dev) [inline lib/screens/settings.dart:84]: image

Please check if this error exist or its in my editor

**Review by dk-a-dev — CHANGES_REQUESTED** [review]
Refacode code as per newly added lint rules
- Use 'package:' imports for files in the 'lib' directory.Try converting the URI to a 'package:' URI.
- Unnecessary use of double quotes.Try using single quotes unless the string contains single quotes.
- Missing a required trailing comma.

Also please resolve conflicts, once code refactoring is done and conflicts are resolved I will review feats implemented 

Feel free to reach out for any sort of help 💖

File: lib/screens/settings.dart (hunk 1 above)
Comment (0kt1) [reply lib/screens/settings.dart:83]: Hey @dk-a-dev, I have updated the changeTheme function in the ThemeProvider screen, which requires a positional argument. This is the reason you are encountering an error.
image
image

**Comment by 0kt1** [comment]
Hey @dk-a-dev , I have resolved the conflicts and refactored the code according to lint rules. Could you please review it? Thanks."

File: lib/screens/settings.dart
Comment (dk-a-dev) [inline lib/screens/settings.dart:1]: back button : colour white in light theme makes it unvisible

File: lib/screens/search_repos.dart
Comment (dk-a-dev) [inline lib/screens/search_repos.dart:1]: also same with settings icon in light mode makes it unvisible

**Review by dk-a-dev — COMMENTED** [review]
Please look for changes suggested, good work
Suggestion: Can u improve UI of the settings tab

File: lib/screens/search_repos.dart
Comment (0kt1) [reply lib/screens/search_repos.dart:1]: ok👍

File: lib/screens/settings.dart
Comment (0kt1) [reply lib/screens/settings.dart:1]: Actually, it's visible in light mode as well, same with the settings icon.

**Comment by dk-a-dev** [comment]
Please pull from origin main once so that ur branch gets update with master commits
this may cause merge conflicts but I will solve them once u pull and iff face issues le me knw

**Comment by 0kt1** [comment]
Hey @dk-a-dev, I am not able to get the master commits. It's showing it's already up-to-date.

**Comment by dk-a-dev** [comment]
@0kt1 sorry for the repo lockdown, we were fixing the issue of repo not showing up for hacktoberfest, its fixed now

**Comment by 0kt1** [comment]
Fine.
@dk-a-dev, can you please review my PR and approve it, or are there any changes that need to be made?

**Comment by dk-a-dev** [comment]
@0kt1 is that settings colour issue fixed, if that fixed this pr will be merged

**Comment by 0kt1** [comment]
@dk-a-dev , Actually, there is no issue with the settings icon, it's changing color in each mode.

File: lib/screens/search_repos.dart [hunk 2]
```diff
@@ -41,6 +39,26 @@ class _SearchState extends State<Search> {
...
+      inputColor = Colors.white;
+    }else if (isDarkMode == false && device.theme == 'System Default'){
+      inputColor = Colors.black;
+    }
+
+
+
     return Scaffold(
       appBar: AppBar(
         backgroundColor: Theme.of(context).primaryColor,
```
Comment (dk-a-dev) [inline lib/screens/search_repos.dart:64]:
```dart
iconTheme: IconThemeData(
          color: Theme.of(context).secondaryHeaderColor,
),

Add this there asap, with more fixes its ready to merge
```

File: lib/screens/search_repos.dart
Comment (dk-a-dev) [inline lib/screens/search_repos.dart:1]: also remove line 66_69
```dart
shape: const BeveledRectangleBorder(
        borderRadius: BorderRadius.all(Radius.circular()),
),
```

File: lib/screens/settings.dart [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+  Map<String, int> themes = {'System Default':1, 'Light':2, 'Dark':3};
+
+
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Theme.of(context).primaryColor,
```
Comment (dk-a-dev) [inline lib/screens/settings.dart:29]:
```dart
iconTheme: IconThemeData(
          color: Theme.of(context).secondaryHeaderColor,
),
```

also remove
```dart
shape: const BeveledRectangleBorder(
        borderRadius: BorderRadius.all(Radius.circular()),
),
```

**Comment by 0kt1** [comment]
@dk-a-dev ,I updated the final fixes.

**Comment by 0kt1** [comment]
@dk-a-dev ,Thank You!

**Pull request merged**